doc/dev/deduplication.rst

```diff
@@ -0,0 +1,254 @@
+  in the target_pool.
+  A redirected object will forward all operations from the client to the 
+  target_object. ::
+  
+        rados -p base_pool set-redirect <base_object> --target-pool <target_pool> 
+         <target_object>
+
+* set-chunk 
+
+  set chunk-offset in a source_object to make a link between it and a 
+  target_object. ::
+  
+        rados -p base_pool set-chunk <source_object> <offset> <length> --target-pool 
+         <caspool> <target_object> <taget-offset> 
+
+* tier-promote 
+
+  promote the object (including chunks). ::
+
+        rados -p base_pool tier-promote <obj-name> 
+
+* unset-manifest
+
+  unset manifest option from the object that has manifest. ::
+
+        rados -p base_pool unset-manifest <obj-name>
+
+
+Dedup tool
+==========
+
+Dedup tool has two features: finding optimal chunk offset for dedup chunking 
+and fixing the reference count.
+
+* find optimal chunk offset
+
+  a. fixed chunk  
+
+    To find out a fixed chunk length, you need to run following command many 
+    times while changing the chunk_size. ::
+
+            ceph-dedup-tool --op estimate --pool $POOL --chunk-size chunk_size  
+              --chunk-algorithm fixed --fingerprint-algorithm sha1|sha256|sha512
+
+  b. rabin chunk(Rabin-karp algorithm) 
+
+    As you know, Rabin-karp algorithm is string-searching algorithm based
+    on a rolling-hash. But rolling-hash is not enough to do deduplication because 
+    we don't know the chunk boundary. So, we need content-based slicing using 
+    a rolling hash for content-defined chunking.
+    The current implementation uses the simplest approach: look for chunk boundaries 
+    by inspecting the rolling hash for pattern(like the
+    lower N bits are all zeroes). 
+      
+    - Usage
+
+      Users who want to use deduplication need to find an ideal chunk offset.
+      To find out ideal chunk offset, Users should discover
+      the optimal configuration for their data workload via ceph-dedup-tool.
+      And then, this chunking information will be used for object chunking through
+      set-chunk api. ::
+
+              ceph-dedup-tool --op estimate --pool $POOL --min-chunk min_size  
+                --chunk-algorithm rabin --fingerprint-algorithm rabin
+
+      ceph-dedup-tool has many options to utilize rabin chunk.
+      These are options for rabin chunk. ::
+
+              --mod-prime <uint64_t>
+              --rabin-prime <uint64_t>
+              --pow <uint64_t>
+              --chunk-mask-bit <uint32_t>
+              --window-size <uint32_t>
+              --min-chunk <uint32_t>
+              --max-chunk <uint64_t>
+
+      Users need to refer following equation to use above options for rabin chunk. ::
+
+              rabin_hash = 
+                (rabin_hash * rabin_prime + new_byte - old_byte * pow) % (mod_prime)
+
+  c. Fixed chunk vs content-defined chunk
+
+    Content-defined chunking may or not be optimal solution.
+    For example,
```

Comment:
could you reconsider the comment at https://github.com/ceph/ceph/pull/28462#discussion_r295818590 ?

Reply:
see the rendered webpage at http://docs.ceph.com/ceph-prs/28462/dev/deduplication/#dedup-tool

Reply:
@tchaikov Sorry, I forgot it. Repushed.